internal/internal_workflow_client.go

```diff
@@ -1515,3 +1542,701 @@ func (w *workflowClientInterceptor) QueryWorkflow(
 
 // Required to implement ClientOutboundInterceptor
 func (*workflowClientInterceptor) mustEmbedClientOutboundInterceptorBase() {}
+
+type scheduleClientInterceptor struct{ client *WorkflowClient }
+
+func (s *scheduleClientInterceptor) CreateSchedule(ctx context.Context, in *ScheduleClientCreateInput) (ScheduleHandle, error) {
+	// This is always set before interceptor is invoked
+	ScheduleID := in.Options.ScheduleID
+	if ScheduleID == "" {
+		return nil, fmt.Errorf("no schedule ID in options")
+	}
+
+	dataConverter := WithContext(ctx, s.client.dataConverter)
+	if dataConverter == nil {
+		dataConverter = converter.GetDefaultDataConverter()
+	}
+
+	action, err := convertToPBScheduleAction(ctx, s.client, in.Options.Action)
+	if err != nil {
+		return nil, err
+	}
+
+	memo, err := getWorkflowMemo(in.Options.Memo, dataConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	searchAttr, err := serializeSearchAttributes(in.Options.SearchAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	var triggerImmediately *schedulepb.TriggerImmediatelyRequest
+	if in.Options.TriggerImmediately {
+		triggerImmediately = &schedulepb.TriggerImmediatelyRequest{
+			OverlapPolicy: in.Options.Overlap,
+		}
+	}
+
+	backfillRequests := convertToPBBackfillList(in.Options.ScheduleBackfill)
+
+	// run propagators to extract information about tracing and other stuff, store in headers field
+	startRequest := &workflowservice.CreateScheduleRequest{
+		Namespace:  s.client.namespace,
+		ScheduleId: ScheduleID,
+		RequestId:  uuid.New(),
+		Schedule: &schedulepb.Schedule{
+			Spec:   convertToPBScheduleSpec(&in.Options.Spec),
+			Action: action,
+			Policies: &schedulepb.SchedulePolicies{
+				OverlapPolicy:  in.Options.Overlap,
+				CatchupWindow:  &in.Options.CatchupWindow,
+				PauseOnFailure: in.Options.PauseOnFailure,
+			},
+			State: &schedulepb.ScheduleState{
+				Notes:            in.Options.Note,
+				Paused:           in.Options.Paused,
+				LimitedActions:   in.Options.RemainingActions != 0,
+				RemainingActions: in.Options.RemainingActions,
+			},
+		},
+		InitialPatch: &schedulepb.SchedulePatch{
```

Comment:
Looking at the server code it's fine, but maybe slightly wasteful (since the server will trigger a patch that does nothing). It is very little work to leave out.